### PR TITLE
New: [AEA-4047] - Set statuses to temporarily unavailable if nppts fails

### PR DESCRIPTION
**Pull request by originalphil**
## Summary

- Routine Change

### Details

No longer use catch all error for nppts call. Always progress to enrich prescriptions. Update items expecting status updates with temporary message.

### CODE REVIEW

File: packages/enrichPrescriptions/src/enrichPrescriptions.ts [hunk 1]
```diff
@@ -29,9 +35,13 @@ export async function lambdaHandler(event: EnrichPrescriptionsEvent) {
...
-  if (statusUpdates) {
+  if (statusUpdates && statusUpdates.isSuccess) {
     logger.info("Applying status updates.")
     applyStatusUpdates(searchsetBundle, statusUpdates)
+  } else if (statusUpdates && !statusUpdates.isSuccess) {
+    logger.info("Call to get status updates was unsuccessful. Applying temporary status updates.")
+    const statusUpdateRequest = event.statusUpdateData!
+    applyTemporaryStatusUpdates(searchsetBundle, statusUpdateRequest)
   } else {
     logger.info("No status updates to apply.")
```
Comment (MatthewPopat-NHS) [inline packages/enrichPrescriptions/src/enrichPrescriptions.ts:46]: We should be saying temporarily unavailable in all cases where we can't apply updates.

File: SAMtemplates/state_machines/GetMyPrescriptionsStateMachine.asl.json [hunk 2]
```diff
@@ -60,7 +60,7 @@
         "FunctionName": "${GetStatusUpdatesFunctionArn}"
       },
       "InputPath": "$.statusUpdateData",
-      "Next": "Get Status Updates Result",
+      "Next": "Enrich Prescriptions",
```
Comment (MatthewPopat-NHS) [inline SAMtemplates/state_machines/GetMyPrescriptionsStateMachine.asl.json:63]: The catch block should also be updated to point to "Enrich Prescriptions".

Comment (originalphil) [reply SAMtemplates/state_machines/GetMyPrescriptionsStateMachine.asl.json:63]: Sorted.

File: packages/enrichPrescriptions/src/enrichPrescriptions.ts (hunk 1 above)
Comment (originalphil) [reply packages/enrichPrescriptions/src/enrichPrescriptions.ts]: Done.

File: packages/getMyPrescriptions/tests/statusUpdate.test.ts [hunk 3]
```diff
@@ -116,11 +122,15 @@ describe("Unit tests for statusUpdate, via handler", function () {
     jest.useFakeTimers()
   })
 
-  it("when event is processed, statusUpdateData is included in the response", async () => {
+  it.skip("when event is processed, statusUpdateData is included in the response", async () => {
```
Comment (MatthewPopat-NHS) [inline packages/getMyPrescriptions/tests/statusUpdate.test.ts:125]: should this be skipped?

Comment (originalphil) [reply packages/getMyPrescriptions/tests/statusUpdate.test.ts]: It should not. Well spotted!

**Review by MatthewPopat-NHS — APPROVED** [review]

**Pull request merged**